app/controllers/api/v1/workflows_controller.rb

```diff
@@ -69,12 +69,28 @@ def extract_strings(tasks)
     [tasks, task_string_extractor.collector]
   end
 
+  def add_relation(resource, relation, value)
+    if relation == :retired_subjects && value.is_a?(Array)
+      value.each {|id| resource.retire_subject(id) }
+    else
+      super
+    end
+  end
+
   def new_items(resource, relation, value)
-    items = construct_new_items(super(resource, relation, value), resource.project_id)
-    if items.any? { |item| item.is_a?(SubjectSet) }
-      items
+    case relation
+    when :retired_subjects, 'retired_subjects'
+      # TODO: scope_for
```

Comment:
This will be nice to have as researchers may want to remove subjects from circulation, see https://github.com/zooniverse/Panoptes/issues/982#issuecomment-112829720